spec/src/main/asciidoc/configuration-class.adoc

```diff
@@ -0,0 +1,237 @@
+values] they may xref:terminology.adoc#resolved-configuration-path[resolve] to.
+
+NOTE: Because a configuration class represents a set of configuration paths, it also represents a portion of a
+xref:terminology.adoc#configuration-domain[configuration domain].
+
+NOTE: A configuration class also comprises part of a xref:terminology.adoc#load-request[load request]. See the
+xref:loading.adoc[Loading] section for more details.
+
+Configuration classes may be defined by a xref:terminoloy.adoc#component-developer[component developer] to represent
+all, or portions, of her component's configuration.
+
+== Structural Requirements
+
+The following are the structural requirements of configuration classes:
+
+. A configuration class must be an {jls}/jls-9.html[interface].
+
+. A configuration class must not declare type parameters, or non-portable behavior results. (A configuration class must
+be a non-generic interface.)
+
+. A configuration class must not be an {jls}/jls-9.html#jls-9.6[annotation interface].
+
+. A configuration class may be a {jls}/jls-8.html#jls-8.5[member interface]. (There are no additional restrictions on
+the nesting kind of a configuration class imposed by this specification.)
+
+. A configuration class must be `public`.
+
+. A configuration class must not be `sealed`.
+
+. A configuration class must not be hidden.
+
+. A configuration class may have any legal name. (There are no additional restrictions on the class name imposed by
+this specification.)
+
+. A configuration class may extend other interfaces, provided that the resulting specialization adheres to all the
+requirements of this section.
+
+. A configuration class may contain `static` methods and `static` fields. (There are no additional restrictions on
+`static` members imposed by this specification.)
+
+. A configuration class may declare member classes and interfaces. (There are no additional restrictions on member
+classes and interfaces imposed by this specification.)
+
+. A configuration class may contain `default` methods.
+
+. A configuration class may contain `private` methods. (There are no additional restrictions on `private` methods
+imposed by this specification.)
+
+. A `public`, non-`default` instance method of a configuration class must not declare type parameters, or non-portable
+behavior results.
+
+. A `public` instance method of a configuration class may declare parameters.
+
+.. If a `public`, non-`default` instance method of a configuration class declares parameters, and the configuration
+class is supplied as part of a load request, a Jakarta Config implementation may choose to either:
+
+... Implement the method somehow, provided that it fulfils all other structural requirements, or
+
+... Implement the method such that it throws either:
+
+.... a `jakarta.config.NoSuchObjectException`, or
+
+.... a `java.lang.UnsupportedOperationException`
```

Comment:
I think we can simplify this, and just don't allow instance methods with parameters.